src/buildstream/_scheduler/jobs/job.py

```diff
@@ -312,11 +313,26 @@ def child_action(self):
             self.action_name, self._message_element_name, self._message_element_key
         )
 
-        # Time, log and and run the action function
-        #
-        with self._messenger.timed_suspendable() as timeinfo, self._messenger.recorded_messages(
-            self._logfile, self._scheduler.context.logdir
-        ) as filename:
+        with ExitStack() as stack:
+            # Time, log and and run the action function
+            #
+            timeinfo = stack.enter_context(self._messenger.timed_suspendable())
+
+            try:
+                filename = stack.enter_context(
+                    self._messenger.recorded_messages(self._logfile, self._scheduler.context.logdir)
+                )
+            except Exception:
+                elapsed = datetime.datetime.now() - timeinfo.start_time
+                self.message(
+                    MessageType.ERROR,
+                    "Error opening log file",
```

Comment:
Could be a bit better to have `except Exception as e` and `"Error opening log file: {}".format(e)` here.

It could be (I'm not altogether sure off the top of my head) that the logging configuration might truncate the "detail" string if it has too many lines, so we *might* not get the entire stack trace there.

Reply:
Pushed this fix myself and merging.